lld/docs/ELF/linker_script.rst

```diff
@@ -198,13 +198,52 @@ the current location to a max-page-size boundary, ensuring that the next
 LLD will insert ``.relro_padding`` immediately before the symbol assignment
 using ``DATA_SEGMENT_RELRO_END``.
 
+Section Classes
+~~~~~~~~~~~~~~~
+
+``SECTIONS`` commands can define classes of input sections:
+
+::
+
+  SECTIONS {
+    CLASS(class_name) {
+      input-section-description
+      input-section-description
+      ...
+    }
+  }
+
+Input section descriptions can refer to a class using ``CLASS(class_name)``
+instead of the usual filename and section name patterns. For example:
+
+::
+
+  SECTIONS {
+    CLASS(c) { *(.rodata.earlier) }
+    .rodata { *(.rodata) CLASS(c) (*.rodata.later) }
```

Comment:
While CLASS(C) { *(.rodata.earlier) } isn't going to clash with any OutputSection description as there is no `:` before the `{`. It is possible that `CLASS(C)` will clash with an existing linker script with an object file called `CLASS` and a section called `c`. We could accept the risk as I think that example is contrived, or perhaps use some different syntax for example `CLASS[c]`

Reply:
`CLASS` could be a file name, conflicting with existing script. But we can accept the risk, and users could use `"CLASS"(.text)` if they want to use `CLASS` as the file name...